File: 13 opencv（四）/c++/pyr_laplace.cpp
#include<opencv2/opencv.hpp>

using namespace cv;

int main() {
    cv::Mat img = cv::imread("1.jpg");

    cv::Mat img_down, img_up, img_new;
    cv::pyrDown(img, img_down);
    cv::pyrUp(img_down, img_up);

    cv::subtract(img, img_up, img_new);

    cv::imshow("img_LP", img_new);
    cv::waitKey(0);
}

File: 13 opencv（四）/pyr_laplace.py
import cv2

img = cv2.imread(r"12.jpg")
print(img.size)
img_down = cv2.pyrDown(img)
print(img_down.size)
img_up = cv2.pyrUp(img_down)
print(img_up.size)
img_new = cv2.subtract(img, img_up)
#为了更容易看清楚，做了个提高对比度的操作
img_new = cv2.convertScaleAbs(img_new, alpha=20, beta=0)
cv2.imshow("img_LP", img_new)
cv2.waitKey(0)